contrib/gogearbox/gearbox.v1/gearbox_test.go

```diff
@@ -0,0 +1,280 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package gearbox
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/gogearbox/gearbox"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/mocktracer"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Test that the gearbox request context retains the tracer context
+// func TestChildSpan(t *testing.T) {
+// 	assert := assert.New(t)
+// 	mt := mocktracer.Start()
+// 	defer mt.Stop()
+
+// 	go newGbServer()
+
+// 	client := &http.Client{
+// 		Timeout: 3 * time.Second,
+// 	}
+// 	_, err := client.Get("http://127.0.0.1:1234/any")
+// 	require.Equal(t, nil, err)
+
+// 	_, ok := tracer.SpanFromContext(reqctx)
+// 	assert.True(ok)
+// }
+
+// func TestPropagation(t *testing.T) {
+// 	t.Run("inject-extract", func(t *testing.T) {
+// 		assert := assert.New(t)
+// 		mt := mocktracer.Start()
+// 		defer mt.Stop()
+
+// 		fcc := &gearboxutil.FastHTTPHeadersCarrier{}
+
+// 		go newGbServer()
+
+// 		client := &http.Client{
+// 			Timeout: 3 * time.Second,
+// 		}
+// 		_, err := client.Get("http://127.0.0.1:1234/any")
+// 		require.Equal(t, nil, err)
+// 		spans := mt.FinishedSpans()
+
+// 		gb := &GearboxContextMock{requestCtx: reqctx}
+
+// 		pspan := tracer.StartSpan("test")
+// 		err := tracer.Inject(pspan.Context(), fcc)
+// 		if err != nil {
+// 			t.Fatalf("Trace injection failed")
+// 		}
+// 		Middleware(WithServiceName("gb"))(gb)
+// 		sctx, err := tracer.Extract(fcc)
+// 		if err != nil {
+// 			t.Fatalf("Trace extraction failed")
+// 		}
+// 		assert.Equal(sctx.TraceID(), pspan.Context().TraceID())
+// 		assert.Equal(sctx.SpanID(), pspan.Context().SpanID())
+// 	})
+// 	t.Run("req-context", func(t *testing.T) {
+// 		assert := assert.New(t)
+// 		mt := mocktracer.Start()
+// 		defer mt.Stop()
+
+// 		reqctx := newReqCtx(200)
+// 		gb := &GearboxContextMock{requestCtx: reqctx}
+// 		fcc := &gearboxutil.FastHTTPHeadersCarrier{ReqHeader: &gb.Context().Request.Header}
+
+// 		pspan := tracer.StartSpan("test")
+// 		err := tracer.Inject(pspan.Context(), fcc)
+// 		if err != nil {
+// 			t.Fatalf("Trace injection failed")
+// 		}
+// 		Middleware(WithServiceName("gb"))(gb)
+// 		span, ok := tracer.SpanFromContext(gb.Context())
+// 		assert.True(ok)
+// 		assert.Equal(span.(mocktracer.Span).TraceID(), pspan.(mocktracer.Span).TraceID())
+// 		assert.Equal(span.(mocktracer.Span).ParentID(), pspan.(mocktracer.Span).SpanID())
+// 	})
+// }
+
```

Comment:
Every test is successful when I run them isolated. But if I run all package tests sequentially (e.g, with "run package tests" vscode feature), TestWithStatusCheck, TestCustomResourceName and TestIgnoreResources fail.